tests/enum_input.rs

```diff
@@ -0,0 +1,133 @@
+use rand::{prelude::ThreadRng, thread_rng, Rng};
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+use std::net::SocketAddr;
+
+use ggrs::{Config, Frame, GGRSRequest, GameStateCell, InputStatus};
+
+fn calculate_hash<T: Hash>(t: &T) -> u64 {
+    let mut s = DefaultHasher::new();
+    t.hash(&mut s);
+    s.finish()
+}
+
+pub struct GameStub {
+    pub gs: StateStub,
+}
+use bytemuck::{CheckedBitPattern, NoUninit, Zeroable};
+
+#[repr(u8)]
+#[derive(Copy, Clone, PartialEq, CheckedBitPattern, NoUninit)]
+pub enum EnumInput {
+    Val1,
+    Val2,
+}
+
+unsafe impl Zeroable for EnumInput {
+    fn zeroed() -> Self {
+        unsafe { core::mem::zeroed() }
+    }
+}
+
+pub struct StubConfig;
+
+impl Config for StubConfig {
+    type Input = EnumInput;
+    type State = StateStub;
+    type Address = SocketAddr;
+}
+
+impl GameStub {
```

Comment:
I wrapped up tests in a very ugly fashion, just as a minimal check, I gotta clean this up, we probably should make a generic test over any input, and test over regular struct, and enums.